game/System/CoreServices/Boot.aoscservice/ASBootloaderCoreService.rpy

```diff
@@ -17,11 +17,30 @@ init 5 python:
         bundleDescription = """\
             Show a loading screen for AliceOS.
         """
+
+        # Looks for all apps using AppKit and returns a list of them.
+        def gatherAllApplications(self):
+            import gc
+            apps = []
+            for obj in gc.get_objects():
+                if isinstance(obj, ASAppRepresentative):
+                    apps.append(obj)
+            return apps
         
-        def boot(self, timeout=5, expressSetup=True, disclaimer=None):
-            renpy.call_screen("ASBootloaderView", timeout=timeout)
+        def boot(self, timeout=5, expressSetup=True, disclaimer=None, bootView="ASBootloaderView"):
             if not persistent.AS_COMPLETED_SETUP:
                 ASSetup.startSetup(express=expressSetup, disclaimer=disclaimer)
+
+            for app in self.gatherAllApplications():
```

Comment:
might be better to thread startup applications so it doesn't slow down the whole process.

Reply:
So probably have a thread for each `applicationWillLaunchAtLogin`?

Reply:
no, run a seperate init thread per startup app, this is considerably safer.